src/render.lisp

```diff
@@ -170,6 +170,35 @@ about the specified renderer, and return it."
 (defun get-renderer-output-size (renderer)
   (niy "SDL_GetRendererOutputSize()"))
 
+(defun query-texture (texture &key format access width heigth)
+  (with-foreign-objects ((format-ptr :uint32)
+			 (access-ptr :int)
+			 (width-ptr  :int)
+			 (heigth-ptr :int))
+    (check-rc (sdl-query-texture texture format-ptr access-ptr width-ptr heigth-ptr))
+    (values (mem-ref format-ptr :uint32)
+	    (mem-ref access-ptr :int)
+	    (mem-ref width-ptr  :int)
+	    (mem-ref heigth-ptr :int))))
+
+;;; Convenience functions to query only textures width and height
+(defun texture-width (texture)
+  (with-foreign-object (width-ptr :int)
+    (check-rc (sdl-query-texture texture
+				 (cffi:null-pointer)
+				 (cffi:null-pointer)
+				 width-ptr
+				 (cffi:null-pointer)))
+    (mem-ref width-ptr :int)))
+
+(defun texture-height (texture)
+  (with-foreign-object (height-ptr :int)
+    (check-rc (sdl-query-texture texture
+				 (cffi:null-pointer)
+				 (cffi:null-pointer)
+				 (cffi:null-pointer)
+				 height-ptr))
+    (mem-ref height-ptr :int)))
```

Comment:
Likewise with `C-WITH`.  Also, you can just use `NIL`, and not explicitly specify `(cffi:null-pointer)` with autowrap functions.